agent.py

```diff
@@ -214,7 +217,53 @@ def _select_action(self,
         return action
 
 
-class PieceCountHeuristicsAgent(BaseTreeSearchAgent):
+class BaseMinMaxSearchAgent(BaseTreeSearchAgent):
+    def __init__(self, *required_kwarg_names: str) -> None:
+        super().__init__('depth', *required_kwarg_names)
+
+    def _select_action(self, game: Game, **kwargs) -> Action:
+        _, best_action = self.minimax(game, kwargs['depth'])
+        assert best_action is not None
+        return best_action
+
+    def minimax(self, game: Game, depth: int, alpha: float = -math.inf, beta: float = math.inf) \
```

Comment:
Changed alpha and beta to default to -inf and inf when started, instead of -1000 and 1000 which it was first